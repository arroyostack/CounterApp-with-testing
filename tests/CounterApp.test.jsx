import { fireEvent, getByRole, getByText, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/CounterApp";

describe( 'Test in <CounterApp />', () => {
    const countStart = 20;
    test( 'should math the snapshot', () => {
        const { container } = render( <CounterApp value={ countStart } /> );
        expect( container ).toMatchSnapshot();
    } );

    test( 'should show initial value 20 <CounterApp={20}/>', () => {
        render( <CounterApp value={ 20 } /> );
        expect( screen.getByText( 20 ) ).toBeTruthy();
    } );

    test( 'should increment with btn +1', () => {
        render( <CounterApp value={ countStart } /> );
        fireEvent.click( screen.getByText( '+1' ) );
        expect( screen.getByText( '21' ) ).toBeTruthy();

    } );
    test( 'should decrement with btn -1', () => {
        render( <CounterApp value={ countStart } /> );
        fireEvent.click( screen.getByText( '-1' ) );
        expect( screen.getByText( '19' ) ).toBeTruthy();

    } );
    test( 'should reset with btn reset', () => {
        render( <CounterApp value={ countStart } /> );
        fireEvent.click( screen.getByText( '-1' ) );
        fireEvent.click( screen.getByText( '-1' ) );
        fireEvent.click( screen.getByRole( 'button', { name: 'btn-reset' } ) );
        expect( screen.getByText( '20' ) ).toBeTruthy();

    } );
} );