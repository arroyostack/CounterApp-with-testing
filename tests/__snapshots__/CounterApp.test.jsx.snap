// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <CounterApp /> should math the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
    20
  </h2>
  <button
    aria-label="btn-add"
  >
    +1
  </button>
  <button
    aria-label="btn-sub"
  >
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
